build/yamls/antrea-windows-with-ovs.yml

```diff
@@ -4,39 +4,30 @@ data:
     $ErrorActionPreference = "Stop"
     $mountPath = $env:CONTAINER_SANDBOX_MOUNT_POINT
     $mountPath = ($mountPath.Replace('\', '/')).TrimEnd('/')
-    $OVSDriverDir = "$mountPath\openvswitch\driver"
-
-    # Check if OVSExt driver is already installed
-    $driverStatus = netcfg -q ovsext
-    if ($driverStatus -like '*not installed*') {
-      # Install OVS Driver
-      $result = netcfg -l $OVSDriverDir/ovsext.inf -c s -i OVSExt
-      if ($result -like '*failed*') {
-        Write-Host "Failed to install OVSExt driver: $result"
-        exit 1
-      }
-      Write-Host "OVSExt driver has been installed"
-    }
-
-    # Check if the VC redistributable is already installed.
-    $OVSRedistDir="$mountPath\openvswitch\redist"
-    if (Test-Path $OVSRedistDir) {
-      $dllFound = $false
-      $paths = $env:PATH -split ';'
-      foreach ($path in $paths) {
-        $dllFiles = Get-ChildItem -Path $path -Filter "vcruntime*.dll" -File -ErrorAction SilentlyContinue
-        if ($dllFiles.Count -gt 0) {
-          $dllFound = $true
-          break
-        }
-      }
-
-      # vcruntime dlls are not installed on the host, then install the binaries.
-      if (-not $dllFound) {
-        Get-ChildItem $OVSRedistDir -Filter *.exe | ForEach-Object {
-          Start-Process -FilePath $_.FullName -Args '/install /passive /norestart' -Verb RunAs -Wait
-        }
+    $OVSHelperScript = $mountPath\k\antrea\OVSHelper.psm1
+    if (Test-Path $OVSHelperScript) {
+      Import-Module $OVSHelperScript
+
+      $OVSDriverDir = "$mountPath\openvswitch\driver"
+      # Check if OVSExt driver is already installed. If not installed, or the installed version is different,
+      # install the OVSext driver provided in the container.
+      CheckAndInstallOVSDriver -OVSDriverPath $OVSDriverDir
+
+      $OVSRedistDir="$mountPath\openvswitch\redist"
+      # Install VC redistributable files if they are provided. Directory "openvswitc\redist"
+      # may not exist in the earlier versions of the images, as the required dlls were already
+      # installed into the container images. After using hpc as the base image, VC redistributable
+      # exe file is provided in the image and expects to install on the Windows host.
+      if (Test-Path $OVSRedistDir) {
+        # $mininalVersion is the minimal version required by the provided Windows OVS binary. If a higher
+        # version of VC redistributable file exists on the Windows host, we can skip the installation.
+        $mininalVersion="14.12.25810"
+        # Check if the VC redistributable is already installed. If not installed, or the installed version
+        # is lower than $minimalVersion, install VC redistributable files provided in the container.
+        CheckAndInstallVCRedists -VCRedistPath $OVSRedistDir -VCRedistsVersion $mininalVersion
       }
+    } else {
+      Write-Host "OVS driver is not installed via the host-process containers, please install it manually."
```

Comment:
When `Test-Path $OVSHelperScript` fails, we can't determine if the driver is installed on the host. I think `please install it manually` might be misleading.

Reply:
Is this what you want?
```
 if(-not (Test-Path $OVSHelperScript)) {
      $driverStatus = netcftg -q ovsext
      if ($driverStatus -like "*not installed*") {
         Write-Host "OVS driver is not installed on the Windows Node, please install it manually.
      }
}
```

If there is no $OVSHelperScript provided in the image, we would ask the user to manually install the driver or update it but not operate from container.